test/replication-luatest/gh_6033_box_promote_demote_test.lua

```diff
@@ -0,0 +1,325 @@
+        replication_timeout = 0.5,
+        replication = {
+            helpers.instance_uri('server_', 1),
+        },
+    })
+
+    local wal_write_count = g.server_1:exec(function(r)
+        box.cfg{replication = {r}}
+        return box.error.injection.get('ERRINJ_WAL_WRITE_COUNT')
+    end, {helpers.instance_uri('server_', 3)})
+
+
+    g.server_3:exec(function()
+        local s = box.schema.create_space('test', {is_sync = true})
+        s:create_index('pk')
+        require('fiber').create(s.replace, s, {1})
+    end)
+    g.server_3:wait_synchro_queue_owner()
+    g.server_1:wait_wal_write_count(wal_write_count + 1)
+
+    g.server_3:drop()
+
+    local ok, err = g.server_1:exec(function()
+        box.cfg{replication_synchro_timeout = 0.1}
+        return pcall(box.ctl.promote)
+    end)
+
+    g.server_2:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    print(ok, err)
+    luatest.assert(
+        not ok and err.code == box.error.QUORUM_WAIT,
+        'wait quorum failure not handled')
+end
+
+-- Test interfering demotion for election_mode = 'off'
+-- while in WAL delay
+g.test_wal_interfering_demote = function(g)
+    local wal_write_count = g.server_3:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', true)
+        return box.error.injection.get('ERRINJ_WAL_WRITE_COUNT')
+    end)
+
+    g.server_2:exec(function()
+        box.ctl.promote()
+    end)
+    g.server_2:wait_synchro_queue_owner()
+    g.server_3:wait_wal_write_count(wal_write_count + 1)
+
+    local ok, err = g.server_3:exec(function()
+        local f = require('fiber').new(box.ctl.demote); f:set_joinable(true)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        return f:join()
+    end)
+    luatest.assert(
+        not ok and err.code == box.error.INTERFERING_PROMOTE,
+        'interfering demote not handled')
+end
+
+-- Test interfering demotion for election_mode = 'off'
+-- while in txn_limbo delay
+g.test_limbo_empty_interfering_demote = function(g)
+    local wal_write_count, f = g.server_3:exec(function()
+        box.error.injection.set('ERRINJ_TXN_LIMBO_EMPTY_DELAY', true)
+        local f = require('fiber').new(box.ctl.demote); f:set_joinable(true)
+        return box.error.injection.get('ERRINJ_WAL_WRITE_COUNT'), f:id()
+    end)
+
+    g.server_2:exec(function()
+        box.ctl.promote()
+    end)
+    g.server_2:wait_synchro_queue_owner()
+    g.server_3:wait_wal_write_count(wal_write_count + 1)
+
+    local ok, err = g.server_3:exec(function(f)
+        box.error.injection.set('ERRINJ_TXN_LIMBO_EMPTY_DELAY', false)
+        return require('fiber').find(f):join()
+    end, {f})
+    luatest.assert(
+        not ok and err.code == box.error.INTERFERING_PROMOTE,
+        'interfering demote not handled')
+end
+
+-- Test that box_demote will return 0 when demoting raft leader
+g.test_raft_leader_demote = function(g)
+    local ok, _ = g.server_1:exec(function()
```

Comment:
No need for `_`. Just drop it. In Lua it is not necessary to have the same number of variables on the left side as there are values on the right side. The same in other similar places.